src/navigator_data_ingest/base/utils.py

```diff
@@ -0,0 +1,116 @@
+import json
+import logging
+import os
+from typing import Generator, List, Tuple
+from typing import cast
+
+from cloudpathlib import CloudPath, S3Path
+
+from navigator_data_ingest.base.api_client import (
+    API_HOST_ENVVAR,
+    MACHINE_USER_EMAIL_ENVVAR,
+    MACHINE_USER_PASSWORD_ENVVAR,
+)
+from navigator_data_ingest.base.types import (
+    DocumentGenerator,
+    Update,
+    InputData,
+    Document,
+)
+
+REQUIRED_ENV_VARS = [
+    API_HOST_ENVVAR,
+    MACHINE_USER_EMAIL_ENVVAR,
+    MACHINE_USER_PASSWORD_ENVVAR,
+]
+ENV_VAR_MISSING_ERROR = 10
+
+_LOGGER = logging.getLogger(__file__)
+
+
+class LawPolicyGenerator(DocumentGenerator):
+    """
+    A generator of:
+
+    - Validated new Document objects for inspection & upload.
+    - Validated Update objects for updating existing documents.
+    """
+
+    def __init__(self, input_file: S3Path, output_location_path: S3Path):
+        """Initialize the generator."""
+        _LOGGER.info("Initializing LawPolicyGenerator")
+        json_data = read_s3_json_file(input_file)
```

Comment:
Is the backend going to create this file with "new_documents" and "updated_documents" file references?

Reply:
Yes as discussed see the type here: 

https://github.com/climatepolicyradar/navigator-data-ingest/blob/feature/input-document-with-updates/src/navigator_data_ingest/base/types.py#L208